src/linux_mcp_server/connection/ssh.py

```diff
@@ -164,32 +164,60 @@ async def execute_remote(
         command: list[str],
         host: str,
         username: str,
+        timeout: int | None = None,
```

Comment:
This could default to using `CONFIG.timeout` instead of having to later check the value.